torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -152,4 +152,34 @@ std::vector<xla::XlaOp> CreateBroadcastTensors(
   return result;
 }
 
+xla::XlaOp CreateIndex(const xla::XlaOp& input, const xla::XlaOp& indices) {
```

Comment:
Given index() is scarce of documentation, I'd grab some from Indexing.cpp and add here.

Reply:
Seems also we are not handling some of the cases and "notes" listed in such documentation.
Like null tensors not being indexed and not adjacent indices.
The documentation is not exactly 100% clear either.

Reply:
Also Byte tensor expansion with nonzero() is trouble, as nonzero() means dynamic shapes.

Reply:
I believe at least some of the checks and corner case handling happens before calling into the method we're supposed to implement, but will look into it more.

Why does `nonzero` mean dynamic shapes? It's a simple element-wise not equals.

Reply:
The shape returned by it, depends on the tensor content, not shape:

```
>>> import torch
>>> x = torch.randn(4,4).le(0.5).nonzero()
>>> print(x)
tensor([[0, 0],
        [0, 1],
        [0, 2],
        [0, 3],
        [1, 0],
        [1, 1],
        [2, 0],
        [2, 1],
        [2, 2],
        [3, 0],
        [3, 1],
        [3, 3]])
>>> x = torch.randn(4,4).le(0.5).nonzero()
>>> print(x)
tensor([[0, 1],
        [0, 3],
        [1, 0],
        [1, 2],
        [1, 3],
        [2, 0],
        [2, 1],
        [2, 3],
        [3, 0],
        [3, 3]])
```

Reply:
Ok, this is awful. For the purpose of this patch, running `nonzero` through the ATen interop first and then finishing the indexing through XLA should be possible. We'll have to find a solution for `nonzero` if it's operating on big tensors for real models. A physical, fixed size representation with an additional long scalar to represent the effective size should work.